traits/trait_types.py

```diff
@@ -2810,11 +2810,7 @@ def validate_implements(value, klass, unused=None):
     """ Checks to see if a specified value implements the instance class
         interface (if it is an interface).
     """
-
-    from .has_traits import isinterface
-    from .interface_checker import check_implements
-
-    return isinterface(klass) and check_implements(value.__class__, klass)
+    return isinstance(value, klass)
```

Comment:
Based on my understanding of the ctraits internals, one of the reasons why `validate_implements` needs to be injected into `ctraits.c` is because it uses `isinterface` and `check_implements`.

Based on this change, it looks like we can remove that extra level of complexity and move `validate_implements` entirely into `ctraits.c`.

What do you think?

(I don't have comments on the code changes in this PR ATM)

Reply:
Yes, that should be removable; I wanted to keep this PR focused on the change itself. Once the change is in (*if* it goes in), there's lots of non-behaviour-changing cleanup we can do